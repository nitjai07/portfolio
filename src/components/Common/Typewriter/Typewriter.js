import React, { Component } from 'react';
import PropTypes from 'prop-types';

class Typewriter extends Component {
  constructor(props) {
    super(props);

    this.state = {
      ind: 0,
      speed: 70
    };
  }

  componentDidMount() {
    this.typeWriter();
  }

  componentWillUnmount() {
    clearInterval(this.testInt);
  }

  typeWriter = () => {
    if (this.state.ind < this.props.typeTxt.length) {
      document.getElementById(this.props.idValue).innerHTML += this.props.typeTxt.charAt(this.state.ind);
      this.setState(prevState => ({
        ind: prevState.ind + 1
      }));
      this.testInt = setTimeout(this.typeWriter, this.state.speed);
    }
  };

  render() {
    return <span id={this.props.idValue} />;
  }
}

Typewriter.propTypes = {
  typeTxt: PropTypes.string.isRequired,
  idValue: PropTypes.string.isRequired
};

export default Typewriter;
