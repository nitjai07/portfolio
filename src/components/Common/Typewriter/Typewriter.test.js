import React from 'react';
import { shallow } from 'enzyme';
import Typewriter from './Typewriter';

test('Typewriter Component renders correctly', () => {
  const component = shallow(<Typewriter typeTxt="test text" idValue="testId" />);
  expect(component).toMatchSnapshot();
});
