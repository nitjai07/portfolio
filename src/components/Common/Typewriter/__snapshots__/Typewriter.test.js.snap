// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typewriter Component renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <span
    id="testId"
/>,
  "nodes": Array [
    <span
      id="testId"
/>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_context": Object {},
    "_element": <Typewriter
      idValue="testId"
      typeTxt="test text"
/>,
    "_forcedUpdate": false,
    "_instance": Typewriter {
      "context": Object {},
      "props": Object {
        "idValue": "testId",
        "typeTxt": "test text",
      },
      "refs": Object {},
      "state": Object {
        "ind": 0,
        "speed": 70,
      },
      "typeWriter": [Function],
      "updater": Updater {
        "_callbacks": Array [],
        "_renderer": [Circular],
      },
    },
    "_newState": null,
    "_rendered": <span
      id="testId"
/>,
    "_rendering": false,
    "_updater": Updater {
      "_callbacks": Array [],
      "_renderer": [Circular],
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Typewriter
    idValue="testId"
    typeTxt="test text"
/>,
}
`;
